### DISCUSSION TITLE
Empty value without Optional or ?

### DISCUSSION THREAD

**Discussion by dbpolito** [spatie/laravel-data#482] [General]
Current it seems there is no way to make a field optional without using the `|Optional` class... or the `?` optional flag on property.

I would like to validate an empty array, so:

```php
class Class extends Data
{
    public function __construct(
        public array $values,
    ) {
    }
```

I would like to change this to use `present` instead of `required`, but the`#[Present]` doesn't work, the required is still applied.

Changing to `array|Optional` works but then it will get optional object instead of empty array...

It there a better way to allow an empty array instead of overwriting rules?

```php
    public static function rules(): array
    {
        return [
            'values' => ['array'],
        ];
    }
```

To be honest i thought `present` would be implied by default instead of `required`.

-----

For now what i did was i created a `PresentRuleInferrer` and removed `RequiredRuleInferrer`, maybe that's a good approach, i can open a PR with that if you guys think it's a good approach.

**Comment by rubenvanassche** [comment]
Using present seems correct to me, this was a bug which should be fixed with the release later today.

**Comment by serious-angel** [comment]
Just in case, nowadays, the required is still the default, but the `Present` rule attribute works:

```php
namespace App\Data\Settings;

use App\Enums\Color;
use Spatie\LaravelData\Attributes\MergeValidationRules;
use Spatie\LaravelData\Attributes\Validation\Present;
use Spatie\LaravelData\Data;

#[MergeValidationRules]
final class SomeData extends Data
{
    public function __construct(
        public bool $enabled,
        public ?Color $color,

        /** @var \App\Enums\ColorShade[] */
        #[Present]
        public array $shades,
    ) {}

    public static function rules(): array
    {
        return [
            'shades.*' => [
                'distinct',
            ],
        ];
    }
}
```

Without the attribute, a request with `[]` for `shades` does indeed end up with `HTTP 400`.
